crates/bevy_ecs/src/event.rs

```diff
@@ -323,9 +328,16 @@ impl<T: Component> Events<T> {
         }
     }
 
-    /// A system that calls [Events::update] once per frame.
-    pub fn update_system(mut events: ResMut<Self>) {
-        events.update();
+    /// A system that calls [Events::update] once per frame on all events of the specified type.
+    ///
+    /// This clears all events from the previous frame,
+    /// and advances the current frame's buffer so it is ready to be cleared.
+    pub fn update_system(
+        mut resource_events: ResMut<Self>,
+        mut component_events: Query<&mut Self>,
```

Comment:
Maybe `entity_events` is a more suitable name here, since, as far as I understand, "component" is quite a general term, that can include the concept of "resource" and "event" (which is a resource). Maybe this is not a concern for the public interface, but internally Events are Resources, and Resources are Components.